lib/extensions/ar_region.rb

```diff
@@ -176,4 +176,17 @@ def compressed_id
   def split_id
     self.class.split_id(id)
   end
+
+  def unique_within_region(field, match_case: true)
+    field_value = send(field)
+    field_matches =
+      if match_case
+        self.class.where(field => send(field))
+      else
+        self.class.where("LOWER(#{field}) = ?", field_value.downcase)
+      end
+    unless field_matches.in_region(region_id).where.not(:id => id).empty?
+      errors.add(field, "is not unique within region #{region_id}")
+    end
+  end
```

Comment:
> Is that all extracted from the class name?

Yup. The docstring [here](https://github.com/rails/rails/blob/48f3e702b9687fb33ddf7257fc9064bcb948165e/activemodel/lib/active_model/validations/validates.rb#L6-L125) should assist you more. See lines 115 and 118 on the actual class name extraction you're looking for.